Chore: time and date format refactoring

jrmehle opened the pull request:
**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

Pulled out some commonly used date/time formats into Rails' built-in formatter. Was basically searching for repeated uses of `strftime`. In the process also refactored repeated uses of it out into helpers.

jrmehle committed:
Add time formats initilizer

Use Rails standard way to add time formats. These are used as shortcuts when formatting time.

jrmehle committed:
Utilize mdy time format

jrmehle committed:
Refactor stripe account deadline formatting

Originally was going to make the strftime call a time format but this made more logical sense.

jrmehle committed:
Move commonly used file date format into config

jrmehle committed:
Refactor common subject out into helper method

jrmehle committed:
Utilize existing with formatted deadline date

There is a slight difference being introduced in the date. Previously the month and day did not have a space between them, but now they do.

jrmehle committed:
Merge branch 'supporter_level_goal' into jaredmehle/chore/time-and-date-formats

wwahammy commented on config/initializers/time_formats.rb:
```diff
@@ -0,0 +1,2 @@
+Time::DATE_FORMATS[:mdy] = '%m-%d-%Y'
+Time::DATE_FORMATS[:file_date] = '%m-%d-%Y--%H-%M-%S'
```
I didn't realize this was a thing, sweet!

jrmehle replied:
[Documentation here](https://api.rubyonrails.org/classes/DateTime.html#method-i-to_formatted_s).

wwahammy reviewed (approved):
I like it!

wwahammy commented:
It appears some things broke though :frowning:

jrmehle committed:
Fix stripe account mailer test

Interpreted the code wrong in StripeAccountMailer. The deadline gets passed in and not taken from the nonprofit's stripe account. Made a new time format to reduce duplication.

Also got rid of some duplication in the let vars of the spec.

jrmehle committed:
Merge branch 'jaredmehle/chore/time-and-date-formats' of github.com:CommitChange/houdini into jaredmehle/chore/time-and-date-formats

jrmehle commented:
> It appears some things broke though 😦

I misread the code in the `StripeAccountMailer`. Got that fixed up though. Should be good to go now.
